js/controller.js

```diff
@@ -680,9 +702,20 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
     //called when user selects fullscreen icon
     toggleFullscreen: function() {
       this.state.fullscreen = !this.state.fullscreen;
+      // full support, any element
       if(this.state.isFullScreenSupported) {
         Fullscreen.toggle(this.state.mainVideoWrapper.get(0));
-      } else {
+      }
+      // partial support, video element only (iOS)
+      else if (this.state.isVideoFullScreenSupported) {
+        if(this.state.fullscreen) {
```

Comment:
If we move this.state.fullscreen toggle at the bottom of this method, this IF statement will need to be using the opposite !this.state.fullscreen